src/lm.jl

```diff
@@ -156,7 +156,24 @@ function coeftable(mm::LinearModel)
               ["x$i" for i = 1:size(mm.pp.X, 2)], 4)
 end
 
-predict(mm::LinearModel, newx::Matrix) =  newx * coef(mm)
+# Different methods of prediction for the different factorization types
+getR(x::DensePredChol) = UpperTriangular(x.chol.factors)
+getR(x::DensePredQR) = x.qr[:R]
+
+predict((::LinearModel, newx::Matrix)) = newx * coef(mm)
+
+function predict(mm::LinearModel, newx::Matrix, interval_type::Symbol, level = 0.95)
+    prediction = newx * coef(mm)
+    interval_type == :confint || error("only :confint is currently implemented") #:predint will be implemented
+    R = getR(mm.pp)
+
+    tmp = newx * (R\Diagonal(ones(3,3)))
+    tmp = tmp.^2 * (ones(3,1) * deviance(mm)/dof_residual(mm))
+    tval = quantile(TDist(dof_residual(mm)), 1. -level/2)
+    interval = tval * sqrt(tmp)
+
+    hcat(prediction, prediction + interval, prediction - interval)
```

Comment:
Again, if this is written for vector input then it doesn' have to be allocating. Then we can have another version written for matrix input that just loops over the rows and `X` and calls the method for vector.